api/python/provisioner/commands/release/release.py

```diff
@@ -0,0 +1,291 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+import requests
+
+from configparser import ConfigParser
+from pathlib import Path
+
+from typing import Union, Optional, ClassVar
+from urllib.parse import urlparse, ParseResult
+
+from provisioner.vendor import attr
+from provisioner import config, utils, errors
+from provisioner.salt import (
+    local_minion_id
+)
+from provisioner.pillar import (
+    PillarKey,
+    PillarResolver
+)
+
+logger = logging.getLogger(__name__)
+
+
+def release_url_converter(url: Union[str, Path, ParseResult]):
+    if isinstance(url, Path):
+        return urlparse(f"file://{url.resolve()}")
+    elif isinstance(url, str):
+        return urlparse(url)
+    else:
+        return url
+
+
+@attr.s(auto_attribs=True)
+class CortxReleaseUrl:
+    url: Union[str, ParseResult] = attr.ib(
+        converter=release_url_converter,
+        validator=attr.validators.instance_of(ParseResult),
+    )
+
+    def __str__(self) -> str:
+        return self.url.geturl()
+
+    @property
+    def path(self) -> str:
+        return f"{self.url.netloc}{self.url.path}"
```

Comment:
Do we always expect that `self.url.pah` starts with `/` slash symbol? Maybe it is better to leave such a comment to help readers to understand that it is not a mistake.

Reply:
yes, it's how urllib works here, I added the comment